### DISCUSSION TITLE
accept temperature changes while window is open

### DISCUSSION THREAD

**Discussion by CodingGhost** [jmcollin78/versatile_thermostat#273] [General]
Hi,
Is it wanted behaviour, that the thermostat resets its temperature to the value before the window was opened after closing it?
also, is it wanted that its possible to set the mode HEAT, despite the window being open?

Im asking because I have the following situation: Im sleeping with the window open and the sheduler component sets a new temperature in the morning. In case I did not close my window yet, I would want that the thermostat begins to heat to the temperature set by the scheduler immidiately after I close the window. What currently happens is, the scheduler sets the new temperature, nothing happens because the window is open, and when I close it, the temperature gets set to the value it had before opening the window instead of the one set by the scheduler.

Adding to that, the scheduler sets the HEAT mode at some time, if the window is open, the thermostat will heat regardless and ignore the open window... it seems to me this is also not really wanted?

I would love to hear some opinions on this :)
Kind regards,
Jonas

**Comment by jmcollin78** [comment]
Hello @CodingGhost ,

> Is it wanted behaviour, that the thermostat resets its temperature to the value before the window was opened after closing it?

Yes. When the window is detected closed, the VTherm restores its parameter as before.

> is it wanted that its possible to set the mode HEAT, despite the window being open?

Yes. The human always have the priority. If you decide to heat when your window is open, you can do it. That's your choice. Consider a wrong detection, you can also force the VTherm. There is also a feature named bypass window detection that you can use in that case instead of changing the hvac_mode. It is available by clicking on the window icon on the VTherm Card UI (maybe it is not documented)

**Comment by CodingGhost** [comment]
I dont understand? how would I solve my issue and make the scheduler set the correct temperature?

**Reply by jmcollin78** [reply]
I what do you propose to solve your specific case and all others cases for all other users ?

**Reply by CodingGhost** [reply]
I would have suggested to not reset the state after closing the window. or make it configurable, that would be great

**Reply by CodingGhost** [reply]
maybe check if anything changes while window is open, and then when the window gets closed, reset everything and then re-apply the changes that were made?

**Reply by jmcollin78** [reply]
No resetting the state after closing will break all other users behavior.
Adding another configuration parameter (there is already 60 parameters) for just your case seems too much.

I know that everybody wants to change the behavior of the VTherm to implement its own use case, but that is not possible. There is so many way to use it. I cannot answer all requests, else VTherm will be unmaintainable.

I'm sure that, saving the current mode and temperature when window is open and restoring it when the window is closed, is the most regular use case. 

I know it is frustrating but it is just impossible for me to implement all uses cases and add checkbox and configuration parameters each time. Developing for HA, is not my job but just my hobby.

The good news, is that VTherm implements services / events that allow to integrate with a specific environment / use cases. You will have to implement your use case by using the services and events. If an events / services is missing and can be useful for others, I will be glad to implement it.

**Reply by CodingGhost** [reply]
thats perfectly understandable, and im sure I will be able to implement it by myself, I just thought that more people would like this. I think this behaviour makes it impossible to really use the scheduler component. everytime the window is open, the schedulers changes will get reverted. But if this behaviour is wanted, thats ok.

**Reply by jmcollin78** [reply]
If you can develop, what you can do is to overwrite the saved target temperature and the saved hvac mode when window is open, so that the restoration (window close) will finally take those last parameters and not the previous one.
This is a little trick to do but should be possible if you can develop and add unit tests.

**Reply by CodingGhost** [reply]
I think this is a good idea, I will try to do that as soon as I have some time!
